frame/support/src/weights.rs

```diff
@@ -710,6 +710,10 @@ where
 			degree: 1,
 		})
 	}
+
+	fn calc(weight: &Weight) -> Self::Balance {
```

Comment:
I find this API a bit strange, that now `polynomial` still needs to be implemented but it is not really used, right? Nonetheless, looks like a reasonable micro optimization.